content/articles/create-a-weather-app-in-node.js-using-openweathermap-api/index.md

```diff
@@ -0,0 +1,673 @@
+                                    <%= visibility %> meters
+                                </p>
+                            </div>
+                        </div>
+                        <div class="card-deck">
+                            <div class="card card-accent-dark mb-3" style="max-width: 18rem;">
+                                <div class="card-header">Other info</div>
+                                <div class="card-body text-dark">
+                                    <h5 class="card-title">Humidity: </h5>
+                                    <p class="card-text">
+                                        <%= humidity %> g.m-3
+                                    </p>
+                                </div>
+                                <div class="card-body text-dark">
+                                    <h5 class="card-title">Pressure: </h5>
+                                    <p class="card-text">
+                                        <%= pressure %> N·m−2
+                                    </p>
+                                </div>
+                            </div>
+                        </div>
+
+                        <% } %>
+
+                            <% if(error !== null){ %>
+                                <p>
+                                    <%= error %>
+                                </p>
+                                <% } %>
+        </fieldset>
+        </div>parser": "^1.17.2",
+     "dotenv": "^10.0.0",
+     "express": "^4.15.3",
+     "request": "^2.81.0"
+    }
+}
+```
+
+#### server.js
+
+We shall do the following in our code:
+
+- Require node_modules
+These are the ones which we had installed in our application as shown below:
+
+```javascript
+// Require node_modules
+
+const express = require('express');
+const bodyParser = require('body-parser');
+const request = require('request');
+const app = express();
+
+// Configure dotenv package
+
+require('dotenv').config();
+```
+
+- Set up our API KEY, express app, and body-parser configurations, and our javascript template view engine
+
+```javascript
+// Set up our OpenWeatherMap API_KEY
+
+const apiKey = `${process.env.API_KEY}`;
+
+// Setup our express app and body-parser configurations
+// Setup our javascript template view engine
+// we shall serve our static pages from the public directory, it will act as our root directory
+app.use(express.static('public'));
+app.use(bodyParser.urlencoded({ extended: true }));
+app.set('view engine', 'ejs');
+```
+
+- Setup our default display on launch
+
+```javascript
+// Setup our default display on launch
+app.get('/', function(req, res) {
+
+    // It shall not fetch and display any data in the index page
+    res.render('index', { weather: null, error: null });
+});
+```
+
+- Setup post request display
+  Here, we shall use the city passed in the post request and API_KEY in our '.env' file to get the data from the site.
+
+```javascript
```

Comment:
This code snippet is too long. It contains important information within it in form of comments. I suggest, you break it down into size-able chunks, that are easy to consume, and so that the comments can be code explanations.